### ISSUE TITLE
Kotlin code generator generates Record with invalid kotlin syntax when a column is named "value" and <recordsImplementingRecordN/> is true

### ISSUE THREAD

**Issue by andrey-semenov** [jOOQ/jOOQ#10767]
### Expected behavior

Kotlin code generator generates legal kotlin that will be compiled by the kotlin compiler.

### Actual behavior

Kotlin code generator generates illegal kotlin in the Record class where the underlying table has a column named "value". Please see code below:
```kotlin
    override fun value4(value: String?): FileAttributeRecord {
        value = value
        return this
    }
```
And this results in the following compilation error:
```
Val cannot be reassigned
```

My recommendation is to add `this.` when setting the record values in this way:

```kotlin
    override fun value4(value: String?): FileAttributeRecord {
        this.value = value
        return this
    }
```

### Steps to reproduce the problem

Generator configuration (maven):

```xml
          <!-- Generator parameters -->
          <generator>
            <name>org.jooq.codegen.KotlinGenerator</name>
            <database>
              <name>org.jooq.meta.postgres.PostgresDatabase</name>
              <includes>.*</includes>
              <excludes>
                information_schema.*|
                pg_.*|
                postgis.*
              </excludes>
              <!-- Disable table-valued functions (routines with TABLE return types) -->
              <tableValuedFunctions>false</tableValuedFunctions>

              <!-- Custom types -->
              <forcedTypes>
                <!-- PGgeometry -->
                <forcedType>
                  <userType>org.postgis.PGgeometry</userType>
                  <binding>com.x.x.jooq.types.PGgeometryBinding</binding>
                  <types>GEOGRAPHY.*|GEOMETRY.*</types>
                </forcedType>

                <!-- HStore -->
                <forcedType>
                  <userType>com.x.x.jooq.types.HStore</userType>
                  <binding>com.x.x.jooq.types.HStoreBinding</binding>
                  <types>HSTORE</types>
                </forcedType>

                <!-- JSON -->
                <forcedType>
                  <userType>com.fasterxml.jackson.databind.JsonNode</userType>
                  <binding>com.x.x.jooq.types.JacksonJsonNodeBinding</binding>
                  <types>JSON</types>
                </forcedType>

                <!-- JSONB -->
                <forcedType>
                  <userType>com.fasterxml.jackson.databind.JsonNode</userType>
                  <binding>com.x.x.jooq.types.JacksonJsonNodeBinaryBinding</binding>
                  <types>JSONB</types>
                </forcedType>
              </forcedTypes>
                </database>
                <target>
                  <packageName>com.x.x.db</packageName>
                  <directory>${project.basedir}/src/main/kotlin</directory>
                </target>
            </database>

            <generate>
              <relations>true</relations>
              <records>true</records>
              <fluentSetters>true</fluentSetters>
              <immutablePojos>true</immutablePojos>

              <javaTimeTypes>true</javaTimeTypes>
              <validationAnnotations>true</validationAnnotations>

              <routines>false</routines>
              <daos>false</daos>
              <deprecated>false</deprecated>
            </generate>
          </generator>
```

### Versions

- jOOQ: 3.14.0
- Kotlin: 1.14.10
- Database (include vendor): Postgres 9.5
- OS: Ubuntu 20.04
- JDBC Driver (include name if inofficial driver): 42.2.5

**Comment by lukaseder** [comment]
Thanks a lot for your report. I'll look into this right away

**Comment by lukaseder** [comment]
I can reproduce the problem. The key here is to enable:

```xml
<recordsImplementingRecordN>true</recordsImplementingRecordN>
```

**Comment by lukaseder** [comment]
We should simply qualify the column name with `this.value = value`

**Comment by lukaseder** [comment]
Fixed in 3.15.0. Backport scheduled for 3.14.1 (#10770).

Thanks again for your report!
